Services/CheckRightsService/Broker/Consumers/AccessValidatorConsumer.cs

```diff
@@ -0,0 +1,50 @@
+﻿using LT.DigitalOffice.CheckRightsService.Repositories.Interfaces;
+using LT.DigitalOffice.Kernel.AccessValidator.Requests;
+using LT.DigitalOffice.Kernel.Broker;
+using MassTransit;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.CheckRightsService.Broker.Consumers
+{
+    public class AccessValidatorConsumer : IConsumer<IAccessValidatorRequest>
+    {
+        private readonly ICheckRightsRepository repository;
+
+        public AccessValidatorConsumer(ICheckRightsRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        public async Task Consume(ConsumeContext<IAccessValidatorRequest> context)
+        {
+            var request = context.Message;
+            bool hasRights = false;
+
+            try
+            {
+                hasRights = repository.CheckIfUserHasRight(request.UserId, request.RightId);
+            }
+            catch (Exception ex)
+            {
+                await context.RespondAsync<IOperationResult<bool>>(new
+                {
+                    IsSuccess = false,
+                    Errors = new List<string>
+                    {
+                        ex.Message
+                    },
+                    Body = hasRights
+                });
+                return;
+            }
+
+            await context.RespondAsync<IOperationResult<bool>>(new
+            {
+                IsSuccess = true,
+                Body = hasRights
+            });
+        }
```

Comment:
Maybe here u can use OperationResultWrapper?

Reply:
Do I have to? I hate this method. I think this is much clearer to understand and to use.

Reply:
It violates the principles DRY, https://github.com/LT-Students/DigitalOffice/pull/54#discussion_r471047472

Reply:
I understand that someone may not like this construction, but if we decided to use it everywhere within the team, then it is better to do so

Reply:
meh. ok

Reply:
Actually, with a wrapper this looks much cleaner.
Although I have to admit, that this is not an "OperationResult" wrapper per se. It just creates an object.
Maybe wrapper can be modified to really wrap this into IOperationResult<T> -- now that would be cooler.